lib/Git.php

```diff
@@ -286,6 +304,40 @@ public function test_git() {
 		return ($status != 127);
 	}
 
+	/**
+	 * Determine closest parent git repository for a given path
+	 *
+	 * @access  public
+	 * @return  string  the next parent git repo root dir or empty string, if no parent repo found
+	 */
+	public function absolute_git_dir($path) {
+		$descriptorspec = array(
+			1 => array('pipe', 'w'),
+			2 => array('pipe', 'w'),
+		);
+		$pipes = array();
+		$command = Git::get_bin()." rev-parse --absolute-git-dir";
```

Comment:
The command --absolute-git-dir was added in git 2.x. RHEL 7 (which is used by most companies that did not move to RHEL8 yet) still ships git 1.8 (yes I know it's so much outdated).

I changed it to --git-dir and it worked well for me. would it be ok to use this one or to have a check on the git version?

Reply:
@ochurlaud - --git-dir does not work in my environment, since it is finding a top most .git repo in the directory and not the repo on the nearest parent directory. In my case I have a .git repo in my DokuWiki root directory for this kind of test - and if I use rev-parse --git-dir, then it is ignoring the .git in my data/pages namespace and will get to the top most .git repo in the directory tree. Can you verify this also ?

Reply:
@ochurlaud - I have to precise - my answer above was not fully correct. It was based on a test using realpath(<--git-dir result>). This does not work, since --git-dir is behaving different in two cases:

1.  we are commiting a page in a namespace having a .git path - e.g. nspace1/mypage.txt with nspace1/.git
- in this case --git-dir is providing the relative path ".git"
2. we are commiting a page in a sub-namespace - e.g. nspace1/subnspace/mypage.txt
- in this case --gt-dir is providing the absolute path "<DOKU_WIKI_ROOT>/data/pages/nspace1/.git

So in case 1. we would have to prepend $path of the git command to the --git-dir result to get the absolute path.

Can you confirm my findings ?

Reply:
Yes it's what the got documentation states. I'll propose something to fit both needs.

Reply:
@ochurlaud - did commit [5e5a67c](https://github.com/woolfg/dokuwiki-plugin-gitbacked/pull/74/commits/5e5a67c0e8a3db855a5cc7e0af336accd9307e08) for using --git-dir. It works so far on my test cases. Please have a look and check.